File: LeetCode/Python/0912_Sort_an_Array.py
# 912. Sort an Array
# https://leetcode.com/problems/sort-an-array/


class Solution:
    def sortArray(self, nums: list[int]) -> list[int]:
        min_el = min(nums)
        max_el = max(nums)

        count = [0] * (max_el - min_el + 1)

        for n in nums:
            count[n - min_el] += 1

        j = 0
        for i in range(len(count)):
            while count[i] > 0:
                nums[j] = i + min_el
                count[i] -= 1
                j += 1

        return nums

File: LeetCode/JavaScript/0912-Sort-an-Array.js
// 912. Sort an Array
// https://leetcode.com/problems/sort-an-array/

/**
 * @param {number[]} nums
 * @return {number[]}
 */

var sortArray = function (nums) {
  const minEl = Math.min(...nums);
  const maxEl = Math.max(...nums);

  let count = new Array(maxEl - minEl + 1).fill(0);

  for (let n of nums) {
    count[n - minEl]++;
  }

  let j = 0;
  for (let i = 0; i < count.length; i++) {
    while (count[i] > 0) {
      nums[j] = i + minEl;
      count[i]--;
      j++;
    }
  }

  return nums;
};
